Lib9c/Model/Item/ShopItem.cs

```diff
@@ -49,6 +50,33 @@ public class ShopItem
             Costume = costume;
         }
 
+        public ShopItem(
+            Address sellerAgentAddress,
+            Address sellerAvatarAddress,
+            Guid productId,
+            FungibleAssetValue price,
+            long expiredBlockIndex,
+            INonFungibleItem nonFungibleItem
+        )
+        {
+            SellerAgentAddress = sellerAgentAddress;
+            SellerAvatarAddress = sellerAvatarAddress;
+            ProductId = productId;
+            Price = price;
+            ExpiredBlockIndex = expiredBlockIndex;
+            switch (nonFungibleItem)
+            {
+                case ItemUsable itemUsable:
+                    ItemUsable = itemUsable;
+                    Costume = null;
+                    break;
+                case Costume costume:
+                    ItemUsable = null;
+                    Costume = costume;
+                    break;
+            }
+        }
+
```

Comment:
How about making another similar two constructors use this constructor to reduce code duplication and to specific `expiredBlockIndex`?

For instance:
```
        public ShopItem(Address sellerAgentAddress,
            Address sellerAvatarAddress,
            Guid productId,
            FungibleAssetValue price,
            Costume costume)
            : this(sellerAgentAddress, sellerAvatarAddress, productId, price, 0, costume)
        {
        }
```